File: kotlin/src/main/kotlin/Swap Nodes in Pairs.kt
package o.mg

fun swapPairs(head: ListNode?): ListNode? {
    if (head == null) {
        return head
    }
    if (head.next == null) {
        return head
    }
    var next = head.next
    head.next = swapPairs(next!!.next)
    next.next = head
    return next
}

File: go/solutions/swap_nodes_in_pairs.go
package solutions

func SwapPairs(head *ListNode) *ListNode {
	if head == nil {
		return head
	}
	next := head.Next
	if next == nil {
		return head
	}
	temp := head.Val
	head.Val = next.Val
	next.Val = temp
	next.Next = SwapPairs(next.Next)
	return head
}
